src/vega.schema.ts

```diff
@@ -41,14 +43,27 @@ export type VgValueRef = {
   band?: boolean | number
 }
 
-export type UnionedDomain = {
-  fields: VgDataRef[]
+export type DataRefUnionedDomain = {
+  fields: VgDataRef[],
```

Comment:
I wonder if we should ping Jeff about the syntax for this case.  It reads pretty confusing.